osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ClearAlarmRegisterCommandExecutorTest.java

```diff
@@ -0,0 +1,306 @@
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.openmuc.jdlms.AccessResultCode;
+import org.openmuc.jdlms.AttributeAddress;
+import org.openmuc.jdlms.DlmsConnection;
+import org.openmuc.jdlms.SetParameter;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.DlmsObjectConfigService;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.DlmsObjectType;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ConnectionException;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DlmsMessageListener;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.LoggingDlmsMessageListener;
+import org.opensmartgridplatform.dlms.interfaceclass.InterfaceClass;
+import org.opensmartgridplatform.dlms.interfaceclass.attribute.DataAttribute;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActionResponseDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ClearAlarmRegisterRequestDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.OsgpResultTypeDto;
+import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
+
+@ExtendWith(MockitoExtension.class)
+class ClearAlarmRegisterCommandExecutorTest {
+
+  private static final String OBIS_CODE_ALARM_REGISTER_1 = "0.0.97.98.0.255";
+  private static final String OBIS_CODE_ALARM_REGISTER_2 = "0.0.97.98.1.255";
+
+  private static final int CLASS_ID_CLEAR_ALARM_REGISTER = 1;
+
+  private static final Long ALARM_CODE = 0L;
+
+  @Mock private DlmsObjectConfigService dlmsObjectConfigService;
+
+  @Mock private DlmsConnectionManager connectionManager;
+
+  @Mock private DlmsConnection dlmsConnection;
+
+  private DlmsMessageListener dlmsMessageListener;
+
+  @Mock private ClearAlarmRegisterRequestDto dto;
+
+  @Mock private MessageMetadata messageMetadata;
+
+  @Captor private ArgumentCaptor<SetParameter> setParameterArgumentCaptor;
+
+  private ClearAlarmRegisterCommandExecutor executor;
+
+  @BeforeEach
+  void setup() {
+    this.executor = new ClearAlarmRegisterCommandExecutor(this.dlmsObjectConfigService);
+    this.dlmsMessageListener = new LoggingDlmsMessageListener(null, null);
+  }
+
+  @Test
+  void testAsBundleResponse() throws ProtocolAdapterException, IOException {
+    final ActionResponseDto responseDto = this.executor.asBundleResponse(AccessResultCode.SUCCESS);
+    assertThat(responseDto.getResult()).isEqualTo(OsgpResultTypeDto.OK);
+
+    assertThrows(
+        ProtocolAdapterException.class,
+        () -> {
+          this.executor.asBundleResponse(AccessResultCode.OTHER_REASON);
+        });
```

Comment:
I prefer to use an arrange-act-assert (or given-when-then) approach in unit tests. 
In this case you might consider using catchThrowable to achieve this:
```suggestion
    final Throwable actual = 
        catchThrowable(() -> this.executor.asBundleResponse(AccessResultCode.OTHER_REASON));

    assertThat(actual).isInstanceOf(ProtocolAdapterException.class);
```
Furthermore consider moving this to a separate test method, so you will end up with one test method for returning OK on SUCCES, and one for throwing an exception on OTHER_REASON

Reply:
Changed